.github/workflows/test.yml

```
@@ -0,0 +1,25 @@
+name: Run Tests on Push
+
+on:
+  push:
+    branches:
+      - "*"
```

```suggestion
on:
  pull_request:
    branches:
      - main
```
I think it should only run the tests when making a PR to main and not every time a user pushes to any branch?